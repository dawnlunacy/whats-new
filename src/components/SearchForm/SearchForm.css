form {
    height: 50px;
    width: 100 vw;
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: space-around;
}

input {
    height: 30px;
    width: 45%;
    margin-top: 10px;
    background-color: #141527;
    text-align: center;
    color: white;
    font-size: 0.9em;
}

.search-btn {
    height: 30px;
    width: 15%;
    margin-top: 10px;
    border-radius: 50px;
    text-align: center;
    font-size: 0.7em;
    font-weight: bold;
    background-color: #B56BF1;
    color: white;
}