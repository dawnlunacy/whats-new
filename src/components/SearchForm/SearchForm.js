import React, { Component } from 'react';
import './SearchForm.css';

class SearchForm extends Component {
    constructor() {
        super();
        this.state = {
            input: ""
        };
    }
        handleChange = event => {
            this.setState({ [event.target.name]: event.target.value })
        };

        searchNews = event => {
            event.preventDefault();

            const newsTopicToSearch = {
                keyword: this.state.input
            }

            this.props.searchNewsFor(newsTopicToSearch.keyword.toUpperCase());
            this.resetInput();
        };

        resetInput = () => {
            this.setState({input: ''})
        }

    render() {
        return (
            <form>
                <input
                type="text"
                placeholder="Search for news here"
                name="input"
                value={this.state.input}
                onChange={event => this.handleChange(event)}
                />
                <button onClick ={event => this.searchNews(event)}>Search Now</button>
            </form>
        )
    };
};

export default SearchForm;