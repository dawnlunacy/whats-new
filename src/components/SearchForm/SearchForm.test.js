import React from 'react';
import { shallow } from 'enzyme';
import SearchForm from './SearchForm';

describe('SearchForm', () => {
    let wrapper;
    const mockSearchNewsFor = jest.fn();

    beforeEach(() => {
        wrapper = shallow(<SearchForm searchNewsFor={mockSearchNewsFor}/>)
    });
    
    it ('should match the SearchForm snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should set the state based on an event', () => {
        const searchEvent = {
            target: {
                name: "input",
                value: "Chicken"
            }
        }
        wrapper.instance().handleChange(searchEvent);

        expect(wrapper.state('input')).toEqual(searchEvent.target.value)
    })
})