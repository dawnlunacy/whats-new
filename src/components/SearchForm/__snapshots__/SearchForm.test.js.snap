// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm should match the SearchForm snapshot 1`] = `
<form>
  <input
    name="input"
    onChange={[Function]}
    placeholder="Search for news here"
    type="text"
    value=""
  />
  <button
    onClick={[Function]}
  >
    Search Now
  </button>
</form>
`;
